flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1274,30 +1276,46 @@ abstract class CodeGenerator(
       literalType: TypeInformation[_],
       literalCode: String)
     : GeneratedExpression = {
+
+    // mark this expression as a constant literal
+    generateTerm(literalType, literalCode).copy(literal = true)
+  }
+
+  private[flink] def generateSymbol(enum: Enum[_]): GeneratedExpression = {
+    GeneratedExpression(
+      qualifyEnum(enum),
+      "false",
+      "",
+      new GenericTypeInfo(enum.getDeclaringClass))
+  }
+
+  /**
+    * Generates access to a term (e.g. a field) that does not require unboxing logic.
+    *
+    * @param fieldType type of field
+    * @param fieldTerm expression term of field (already unboxed)
+    * @return internal unboxed field representation
+    */
+  private[flink] def generateTerm(
+      fieldType: TypeInformation[_],
+      fieldTerm: String)
+    : GeneratedExpression = {
     val resultTerm = newName("result")
     val nullTerm = newName("isNull")
```

Comment:
If the `nullTerm` is constant, couldn't we say `nullTerm = "false";` and simplify the remaining code?

Reply:
Same is true for `generateNullLiteral()`

Reply:
True, I will simplify the logic there.